index.js

```diff
@@ -102,6 +106,18 @@ AFRAME.registerComponent('teleport-controls', {
     this.hitEntity.setAttribute('visible', false);
 
     if ('collisionEntities' in diff) { this.queryCollisionEntities(); }
+
+    if (oldData.aimEvent !== data.aimEvent) {
```

Comment:
Maybe updatable events are overkill to maintain. Perhaps if an application wants to control it, they can control when they emit the event (e.g., their own `teleportaim` event).

Reply:
Do you mean that the events that the component will be listen to shouldn't be configurable? so they should be fixed like `teleportaim` and `teleportjump` and is the app the responsible to emit them?

Reply:
I mean the events should be configurable, but they don't need to be updatable. Don't have to check if the event changed and re-register event handlers.